### ISSUE TITLE
Popup Player Crash

### ISSUE THREAD

**Issue by Perflyst** [TeamNewPipe/NewPipe#1855]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [x] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ GB
* __Service:__ none
* __Version:__ 0.14.2
* __OS:__ Linux Android 7.1.2 - 25
* __More Information:__ Galaxy S6, LineageOS 14.1, App from F-Droid

I also know you do not want crash logs in issue section but this happens not only on my device.
I know there is xposed installed BUT xposed does NOT touch this app in any way!

The popup player starts and loads the video and when it tries to play the video it ends with a crash.

<details><summary><b>Crash log</b></summary><p>

```
java.lang.RuntimeException: Unable to stop activity {org.schabi.newpipe/org.schabi.newpipe.player.MainVideoPlayer}: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4202)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4258)
	at android.app.ActivityThread.-wrap6(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1557)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference
	at org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)
	at org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)
	at org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)
	at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1290)
	at android.app.Activity.performStop(Activity.java:6859)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4197)
	... 10 more
java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference
	at org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)
	at org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)
	at org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)
	at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1290)
	at android.app.Activity.performStop(Activity.java:6859)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4197)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4258)
	at android.app.ActivityThread.-wrap6(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1557)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)

```
</p></details>
<hr>

**Comment by theScrabi** [comment]
Please update to v0.14.2 And tell us if it works.

**Comment by Perflyst** [comment]
This is v0.14.2 from F-Droid.

**Comment by theScrabi** [comment]
Ok ill check it

**Comment by theScrabi** [comment]
This is weird, for me it works

**Comment by Perflyst** [comment]
Can I send you any further logs or try something else in order to help? I will also try a re-install of the app.

**Comment by gkeegan** [comment]
@theScrabi Start the video in the regular player, then switch it to popup. It crashes for me. Starting popup regularly still works.

**Comment by gkeegan** [comment]
Going from the regular video player to background audio only play crashes it too.

**Comment by gkeegan** [comment]
@Perflyst Is this what you did? Or did you start popup directly from the "about video" page.

**Comment by Perflyst** [comment]
That is what I did. Played video and started popup from there while video was playing

**Comment by theScrabi** [comment]
This should have been fixed before release :(. Can someone help to fix this.

**Comment by beNitinhere** [comment]
@theScrabi I tested on both 0.14.1 and 0.14.2 version. This crash was not there in the previous build. I'm working on it to fix it.

**Comment by theScrabi** [comment]
Thank you

**Comment by beNitinhere** [comment]
@theScrabi you're welcome😊... Well I have been fixed it

**Comment by Perflyst** [comment]
Can you share apk here, please?

**Comment by AnubrataDS** [comment]
The exact same issue is happening with me when trying to run a debug build from dev branch on my cellphone (Android 8.0)...

**Comment by beNitinhere** [comment]
@Perflyst Sure, but this is a debug build. [Here](https://www.dropbox.com/s/xspy13sdj76479i/Newpipe.apk?dl=0) you go
@AnubrataDS Ya, It's was there. I have fixed it. Don't worry, you can check my pull request.

**Comment by AnubrataDS** [comment]
@beNitinhere Ok so should I just copy your changes to my branch ? I'm working on another issue and I'm stuck because of this crash

**Comment by beNitinhere** [comment]
@AnubrataDS  ya sure, You should pull my changes to your branch

**Comment by AnubrataDS** [comment]
@beNitinhere can you tell me how to do that ? Should I set your branch as remote and then pull ?

**Comment by beNitinhere** [comment]
@AnubrataDS I think the best way is to create a new local branch and pull my changes and then merge this with dev branch but before merging, you stash your changes otherwise merging can remove your local changes

**Comment by AnubrataDS** [comment]
@beNitinhere Ok I'll try and let you know

**Comment by beNitinhere** [comment]
@AnubrataDS Ya sure

**Comment by Perflyst** [comment]
Can you please share the pull request or add a commit message in it which closes this issue?

**Comment by beNitinhere** [comment]
@Perflyst I have already shared the PR. PR title is hacking-theme added I have also add a theme in this PR, Thanks

**Comment by theScrabi** [comment]
@beNitinhere please send a pullrequest.

**Comment by beNitinhere** [comment]
@theScrabi I have already send pr actually I have also added a theme so my
PR title commit is hacking theme added #1851plz review it and let me know
if anything is needed.
 Thanks

**Comment by theScrabi** [comment]
You may not fix this in the theme PR please send a separate one.

**Comment by TobiGr** [comment]
Fixed by #1874
